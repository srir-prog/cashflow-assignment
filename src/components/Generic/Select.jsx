import React from "react";
import { useTranslation } from "react-i18next";
const Select = ({ onChange, value, className }) => {
  //For internationalisation.
  const [t] = useTranslation("common");
  return (
    <select className={className} onChange={onChange} value={value}>
      <option value="income" defaultValue>
        {t("bottom.income")}
      </option>
      <option value="expense"> {t("bottom.expense")}</option>
    </select>
  );
};

export default Select;
