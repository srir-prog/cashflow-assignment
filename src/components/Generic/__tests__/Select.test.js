import React from 'react';
import { shallow } from 'enzyme';
import Select from './../Select';
import renderer from 'react-test-renderer';

describe('Input component', () => {
    let component;
    const props = {
        className: 'className',
        value: 'Income',
        onChange: jest.fn()
    }
    beforeEach(() => {
        component = shallow(<Select {...props} />);
    });
    it('should render correctly', () => {
        //Snapshot testing.
        const tree = renderer.create(<Select {...props} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    it('should render <input> tag', () => {
        const selectElement = component.find('select');
        expect(selectElement.length).toBe(1);
    });
});