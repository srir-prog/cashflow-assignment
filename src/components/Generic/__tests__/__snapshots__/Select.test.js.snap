// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
<select
  className="className"
  onChange={[MockFunction]}
  value="Income"
>
  <option
    defaultValue={true}
    value="income"
  >
    bottom.income
  </option>
  <option
    value="expense"
  >
     
    bottom.expense
  </option>
</select>
`;
